lib/config/index.js

```diff
@@ -63,75 +58,75 @@ if (config.ldap.tlsca) {
 }
 
-config.Environment = Environment
+logger.debug(typeof config.get('facebook.clientID') && config.get('facebook.clientSecret'))
 
 // auth method
-config.isFacebookEnable = config.facebook.clientID && config.facebook.clientSecret
-config.isGoogleEnable = config.google.clientID && config.google.clientSecret
-config.isDropboxEnable = config.dropbox.clientID && config.dropbox.clientSecret
-config.isTwitterEnable = config.twitter.consumerKey && config.twitter.consumerSecret
-config.isEmailEnable = config.email
-config.isOpenIDEnable = config.openID
-config.isGitHubEnable = config.github.clientID && config.github.clientSecret
-config.isGitLabEnable = config.gitlab.clientID && config.gitlab.clientSecret
-config.isMattermostEnable = config.mattermost.clientID && config.mattermost.clientSecret
-config.isLDAPEnable = config.ldap.url
-config.isSAMLEnable = config.saml.idpSsoUrl
-config.isOAuth2Enable = config.oauth2.clientID && config.oauth2.clientSecret
-config.isPDFExportEnable = config.allowPDFExport
-
-// Check gitlab api version
-if (config.gitlab && config.gitlab.version !== 'v4' && config.gitlab.version !== 'v3') {
-  logger.warn('config.js contains wrong version (' + config.gitlab.version + ') for gitlab api; it should be \'v3\' or \'v4\'. Defaulting to v4')
-  config.gitlab.version = 'v4'
-}
+config.load({
+  isFacebookEnable: toBooleanConfig(config.get('facebook.clientID') && config.get('facebook.clientSecret')),
+  isGoogleEnable: toBooleanConfig(config.get('google.clientID') && config.get('google.clientSecret')),
+  isDropboxEnable: toBooleanConfig(config.get('dropbox.clientID') && config.get('dropbox.clientSecret')),
+  isTwitterEnable: toBooleanConfig(config.get('twitter.consumerKey') && config.get('twitter.consumerSecret')),
+  isEmailEnable: toBooleanConfig(config.get('email')),
+  isOpenIDEnable: toBooleanConfig(config.get('openID')),
+  isGitHubEnable: toBooleanConfig(config.get('github.clientID') && config.get('github.clientSecret')),
+  isGitLabEnable: toBooleanConfig(config.get('gitlab.clientID') && config.get('gitlab.clientSecret')),
+  isMattermostEnable: toBooleanConfig(config.get('mattermost.clientID') && config.get('mattermost.clientSecret')),
+  isLDAPEnable: toBooleanConfig(config.get('ldap.url')),
+  isSAMLEnable: toBooleanConfig(config.get('saml.idpSsoUrl')),
+  isOAuth2Enable: toBooleanConfig(config.get('oauth2.clientID') && config.get('oauth2.clientSecret')),
+  isPDFExportEnable: toBooleanConfig(config.get('allowPDFExport'))
+})
+
+
 // If gitlab scope is api, enable snippets Export/import
-config.isGitlabSnippetsEnable = (!config.gitlab.scope || config.gitlab.scope === 'api') && config.isGitLabEnable
+config.load({isGitlabSnippetsEnable: toBooleanConfig(!config.get('gitlab').scope || config.get('gitlab').scope === 'api') && config.get('isGitLabEnable')})
```

Comment:
`isGitlabSnippetsEnable` should be `isGitlabSnippetsEnable_d_` for more correct english.